### PR TITLE
Update state management to use Universal ID

### PR DESCRIPTION
**Pull request by hopsoft**
## Reduce potential ISP/Gateway privacy concerns related to user tracking etc.

> [!NOTE] 
> There aren't any inherent privacy issues related to TurboBoost Commands (TBC). TBC **does NOT track** user/visitor data for any purpose. The original cookie and header management simply served to facilitate a better user experience (UX).

This PR updates the "state" mechanics to use the URL and hidden form fields rather than cookies and headers for this purpose. 

TurboBoost state management provides a facility for remembering and restoring the state used to render the last representation of a page or component. It also provides the facility to support client side optimistic updates that we can resolve server side. All of this tooling enables developers to provide a great reactive user experience to visitors with minimal code. 

---

After more discovery and some conversations with AI, I've teased out the following areas that are potential spots that may can appear suspect to naive privacy tools.

- Storing larger values in cookies
- Using too many cookies
- Attaching larger values to HTML request headers
- Using too many custom request headers
- Updating or mutating an HTML meta tag too frequently

This PR addresses each of the above items to de-risk privacy hawks from attaching a false positive flag due to any library activity or operations.

### CODE REVIEW

File: app/javascript/turbo.js
Comment (hopsoft) [inline app/javascript/turbo.js:25]: Previously we always sent state. We may want to consider bringing this back when we put more attention on state management and optimistic updates.

File: test/dummy/config/importmap.rb
Comment (mhenrixon) [inline test/dummy/config/importmap.rb:13]: After merging my [PR](https://github.com/hopsoft/turbo_boost-streams/pull/51), you should be able to use:

```ruby
pin "@turbo-boost/streams", to: "@turbo-boost/streams.js"
```

File: lib/turbo_boost/commands/runner.rb
Comment (mhenrixon) [inline lib/turbo_boost/commands/runner.rb:208]: Is this why elements broke for me perhaps?

**Review by mhenrixon — APPROVED** [review]
As I wrote you in our private conversation, my only problem seems to be with elements that could be expected with such a refactoring. 

Commands seems to be working great for me (I use that individually for some really cool stuff as well).

My PRs should help simplify the setup for testing import maps, but I see nothing obvious in this PR, nor do I have any issues other than with elements. 

It's good to go unless you depend on elements being fixed first.

File: lib/turbo_boost/commands/state.rb
Comment (hopsoft) [reply lib/turbo_boost/commands/state.rb:35]: I'm happy to pair more and walk you though my thinking on this; however, I just wanted to get the stub in place so we can revisit easily and make future improvements.

File: test/state_test.rb
Comment (hopsoft) [reply test/state_test.rb:6]: True in this case... unless we ever want to establish a default baseline state.

File: lib/turbo_boost/commands/state.rb
Comment (hopsoft) [reply lib/turbo_boost/commands/state.rb:50]: Actually I had that to begin with but there's a situation where you may assign false that we want to honor; hence, the explicit nil check vs a falsey one.

File: lib/turbo_boost/commands/runner.rb
Comment (hopsoft) [reply lib/turbo_boost/commands/runner.rb:259]: Perhaps we can follow up with more clear semantic naming for some of these.

File: test/state_test.rb
Comment (julianrubisch) [reply test/state_test.rb:6]: yes, we could define an adapter interface, but as it stands this test would brake if we ever swap the internals

**Review by BuddyLReno — APPROVED** [review]
I don't have enough context to offer any meaningful feedback but I didn't see any obvious logic errors or typos 👍

**Pull request merged**
